src/FakeItEasy.Extensions.ValueTask/FakeItEasy.Extensions.ValueTask.csproj

```diff
@@ -20,9 +20,9 @@
 
   <ItemGroup Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netstandard2.1'))">
     <Compile Include="../FakeItEasy/ValueTaskReturnValueConfigurationExtensions.cs" />
-    <Compile Include="../FakeItEasy/ValueTaskReturnValueConfigurationExtensions.StronglyTyped.cs" />
+    <Compile Include="../FakeItEasy/Generated/CodeGen/CodeGen.ValueTaskReturnValueConfigurationExtensionsGenerator/*.cs" />

```

Comment:
Relying on the generated source is maybe a little gross. Can revisit if you like.

Reply:
Yeah, it's not ideal. I assume it's because you can't reference the generator in this project, because it will also generate the rest of the code, unrelated to `ValueTask`?
Maybe we could do something to _conditionally_ generate the code? e.g. have an assembly-level attribute to specify which code generators we want? Or maybe something in the project's properties, if it's easily accessible from the generator?

Reply:
Okay. Here's an option. Generate overloads based on presence of a type in the being-compiled assembly. Feels like a reasonable thing to make a decision based on. And as a bonus, we can get rid of LACKS_VALUETYPE.